src/apps/backend/modules/access-token/rest-api/access-token-controller.ts

```diff
@@ -9,10 +10,26 @@ export class AccessTokenController {
     req: Request<CreateAccessTokenParams>,
     res: Response,
   ) => {
-    const accessToken = await AccessTokenService.createAccessToken({
-      username: req.body.username,
-      password: req.body.password,
-    });
+    let accessToken: AccessToken;
+
+    if (req.body.username && req.body.password) {
+      accessToken = await AccessTokenService.createAccessTokenByUsernameAndPassword(
+        req.body.password,
+        req.body.username,
+      );
+    } else if (req.body.contactNumber && req.body.otpCode) {
+      const otp = await OtpService.verifyOTP(
+        req.body.contactNumber,
+        req.body.otpCode,
+      );
```

Comment:
Please move this code to `AccessTokenService.createAccessTokenByContactNumber`. The ideal is controller should do minimal work and it should delegate the work to service.